projects/plugins/crm/admin/email/ajax.php

```diff
@@ -141,17 +141,16 @@ function zeroBSCRM_emails_customer_panel() {
 
 	zeroBSCRM_mark_as_read( $threadID );
 
-	global $zbs;
-
-	$e = 0;
+	$e         = 0;
+	$email_ret = array();
 	foreach ( $email as $em ) {
 		$email_ret[ $e ]['the_id']          = $em->ID;
 		$email_ret[ $e ]['date']            = zeroBSCRM_locale_utsToDate( $em->zbsmail_created );
 		$email_ret[ $e ]['zbsmail_subject'] = $em->zbsmail_subject;
 		if ( $em->zbsmail_content == null ) {
 			$email_ret[ $e ]['zbsmail_content'] = __( 'No content was stored for this message', 'zero-bs-crm' );
 		} else {
-			$email_ret[ $e ]['zbsmail_content'] = wp_kses( zeroBSCRM_io_WPEditor_DBToHTML( $em->zbsmail_content ), $zbs->acceptable_html );
+			$email_ret[ $e ]['zbsmail_content'] = $em->zbsmail_content;
```

Comment:
This still needs output escaping (e.g. try adding `<script>alert();</script>` directly in the `wp_zbs_sys_email_hist` table, `zbsmail_content` column).

```suggestion
			$email_ret[ $e ]['zbsmail_content'] = wp_kses( $em->zbsmail_content, $zbs->acceptable_html );
```

Reply:
Ah yes! Thanks for the reminder, never trust the DB, even if saved content is already escaped.